vimPlugins.nvim-treesitter: fix update.py and update documentation (#321535)
NixOS/nixpkgs: doc/languages-frameworks/vim.section.md

Nix expressions for Vim plugins are stored in [pkgs/applications/editors/vim/plugins](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/editors/vim/plugins). For the vast majority of plugins, Nix expressions are automatically generated by running [`nix-shell -p vimPluginsUpdater --run vim-plugins-updater`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/updater.nix). This creates a [generated.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/generated.nix) file based on the plugins listed in [vim-plugin-names](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/vim-plugin-names).

> After running the updater, if nvim-treesitter received an update, also run [`nvim-treesitter/update.py $(nix-build -A vimPlugins.nvim-treesitter)`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/update.py) to update the tree sitter grammars for `nvim-treesitter`.

Reviewer: ty for the fix. If people are going to run the command like you did, let's put the full path anyway. I've rephrased the beginning hoping it is clearer that one shouldn't have to run the treesitter updater but feel free to ignore:
`When the vim updater detects an nvim-treesitter update, it also runs [pkgs/applications/editors/vim/plugins/nvim-treesitter/update.py $(nix-build -A vimPlugins.nvim-treesitter)`] ...`

Author: Is this what you mean?
```suggestion
When the vim updater detects an nvim-treesitter update, it also runs [`nvim-treesitter/update.py $(nix-build -A vimPlugins.nvim-treesitter)`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/update.py) to update the tree sitter grammars for `nvim-treesitter`.
```

Reviewer: perfect
